### ISSUE TITLE
How to train my own data set

### ISSUE THREAD

**Issue by WSTao** [bellonemauro/SVM_pcl_demo#4]
Hi,
I want to train with my own data. Can I input the point cloud data (kitti) to train, or input the data for the extracted features, such as sift?

**Comment by bellonemauro** [comment]
Yes, you should be able to do that as far as the data format fits the svmlib format

**Comment by WSTao** [comment]
Thankyou,If my data is a 3D point cloud, can I extract the extracted 3D features (NARF, SIFT3D) as input, or can I only convert it into a 2D image before I can process it?

**Comment by bellonemauro** [comment]
yes, the classifier is blind, as far as the data is represented as in the /data/example1/ it should work. In another project I have done it using a different format, which is not your format but it should still work.

**Comment by WSTao** [comment]
OK, thank you very much

**Comment by WSTao** [comment]
Hi,
I compiled the pcl library you provided，and saved the trained model and I got an error during the test: Was not able to open file"model.dat"
I compared the content of the model you provided, the format is the same, but the number of data in each row is quite different.
I also used the parameters -t -tc ,Can get the test results, what is the reason for this?

**Comment by bellonemauro** [comment]
I don't really know, but if you have done the procedure in the right way it should work. If the classifier is trained and saved (you can check the model.dat file to contain actual data) it means that it is working. 
If you cannot load, it means that something is weird in the data format, if I were you I would check the standard in the data file for example check "space" or "tab" or "comma"vs"dot" in the numbers. These standard may be e taken from your system default values when the file is saved, but it expect to be "space" and "dot" when you load. I hope this can help.

**Comment by bellonemauro** [comment]
if you see something different, you are free to modify manually the model.dat file by finding and replacing the right symbol

**Comment by WSTao** [comment]
Thank you very much for your patience, I will carefully compare the differences.

**Comment by bellonemauro** [comment]
no problem, let me know if it works

**Comment by WSTao** [comment]
HI,
I found the reason is that the number of data between the parameters "scaling" and "SV" in the model file is far more than the model you provided. I deleted the extra ones and it is no problem.
but I don't quite understand what this parameter means. The source code points to the "svm_node" structure

**Comment by bellonemauro** [comment]
sure, that values depend on the number of features/examples and scaling you have in your model. You have to set them. Here you can find a nice guide https://www.csie.ntu.edu.tw/~cjlin/papers/guide/guide.pdf

including the data format and other things here 
https://www.csie.ntu.edu.tw/~cjlin/libsvm/

after all, the library in pcl is imported from libsvm

**Comment by WSTao** [comment]
I will study it, thank you very much.

**Comment by WSTao** [comment]
Hi,
I use your program for multi-category tasks. The result is that only one type of accuracy is very high, others are very low. I also try to divide into two-category tasks, which is the same problem. What is the reason?

**Comment by bellonemauro** [comment]
Hi, 
I cannot help with this, I think you have to just play with the parameters (changing kernels and its parameters). However, if you are working on perception and segmentation of point clouds you can write me an email with your material, I am very interested in this types of applications (you can easily find my email on the Chalmers University website)

**Comment by WSTao** [comment]
Thank you very much, I am doing the target classification. I found the reason later. The main reason is that the features I extracted are not good. I changed one and the effect is much better.
